dataset/src/main/scala/frameless/functions/FramelessLit.scala

```diff
@@ -0,0 +1,58 @@
+package frameless.functions
+
+import frameless.TypedEncoder
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.codegen._
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal, NonSQLExpression}
+import org.apache.spark.sql.types.DataType
+
+@deprecated("Use `Lit[A]`", "0.10.2")
+case class FramelessLit[A](obj: A, encoder: TypedEncoder[A]) extends Expression with NonSQLExpression {
```

Comment:
How much sense it makes to keep it in the codebase? It is not used anywhere and I doubt users would be interested in that.

Reply:
That's a breaking change, so it should only happen for a major version

Reply:
@cchantep well, I have a feeling that this PR is in general very much binary breaking. It is API compatible though. 
Let me set the MIMA plugin in a separate PR!

Reply:
Removing a public class does break the API.

Reply:
@cchantep changing public API functions signatures is API breaking as well. I'm wondering what would be the MIMA report now.

Reply:
That's a question of compromise. Removing this class is not required to fix the current issues, but would break the API (even if we can consider that the user should never directly use this public class), the others (mainly 1) updated signature are strictly required changes.

Reply:
@cchantep oh yea, but these are public API changes; they break bin compat as well; have you checked the plugin report btw?

Reply:
I see no reasons why we can't make the major release (and it looks like this PR, with or without this legacy class - is bin incompatible); value classes + Spark 3.2.0 could be good enough for the major version release. 

I think @imarios opionion about the versioning scheme is important there as well ¯\_(ツ)_/¯ I don't think all frameless minor releases a bin compatible.